File: js/verify.js
let impl = require('./implementations.js');

let ten_million = 10000000;
let million     = 1000000;
let tree_depth  = 17;

console.log(impl.sum(ten_million));
console.log(impl.sum_vector(impl.alloc_vector(million)));
console.log(impl.sum_list(impl.alloc_list(million)));
console.log(impl.sum_tree(impl.alloc_full_tree(tree_depth)));

File: java/microbenchmarks/src/main/java/org/enso/microbenchmarks/Verify.java
package org.enso.microbenchmarks;

public class Verify {
    public static void main(String... args) {
        int ten_million = 10_000_000;
        int million = 1_000_000;
        long tree_depth = 17;
        System.out.println(Implementations.sum(ten_million));
        System.out.println(Implementations.sumVector(Implementations.allocVector(million)));
        System.out.println(Implementations.sumList(Implementations.allocList(million)));
        System.out.println(Implementations.sumTree(Implementations.allocFullTree(tree_depth)));
    }
}
